lib/fluent/config/yaml_parser/parser.rb

```diff
@@ -0,0 +1,166 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/config/yaml_parser/section_builder'
+
+module Fluent
+  module Config
+    module YamlParser
+      class Parser
+        def initialize(config, indent: 2)
+          @base_indent = indent
+          @config = config
+        end
+
+        def build
+          s = @config['system'] && system_config_build(@config['system'])
+          c = @config['config'] && config_build(@config['config'], root: true)
```

Comment:
Always requires `config` parent for `label`, `source`, `filter`, `match` and `worker` is bother and doesn't compatible with Fluentd's format. It makes hard to understand how to convert.

Reply:
This format limitations is coming from _YAML format specification and its parser simplicity_.
Fluentd config format does not have 1-to-1 correspondence for YAML config format for now.
It is quite hard to implement to arbitrary place handling for `label`, `source`, `filter`, `match` and `worker`. directives.

I didn't find no mono morphic way from Fluentd config format to YAML format.

Reply:
FYI: Fluent Bit uses similar approach to handle data pipeline:

https://github.com/fluent/fluent-bit/pull/4734

Fluent Bit ini like format should use `[INPUT]`, `[OUTOUT]`, `[FILTER]` sections as a top level settings on the original config implementation but its YAML format does not use them as top level objects.

Reply:
And we already prepared from Fluentd configuration format to YAML format translation documents here:
https://github.com/fluent/fluentd-docs-gitbook/pull/402

Reply:
Our assumed use-case is:
This format is mainly used from Golang's struct Marshal/Unmarshal not by writing by hands.

Reply:
I got it. For example, the following is invalid as yaml:
```yml
- label:
    $name: "@FLUENT_LOG"
   - match:
       $type: "stdout"
       $tag: tag1.**
  - match:
      $type: "null"
      $tag: tag2.**
```